packages/react/src/useAcceptInvitation.ts

```diff
@@ -0,0 +1,46 @@
+import { gql } from 'graphql-request'
+import { useCallback, useContext, useState } from 'react'
+import invariant from 'ts-invariant'
+import { LiteflowContext } from './context'
+import { ErrorMessages } from './errorMessages'
+
+gql`
+  mutation AcceptInvitation($id: UUID!) {
+    acceptInvitation(input: { id: $id }) {
+      invitation {
+        id
+      }
+    }
+  }
+`
+
+export default function useAcceptInvitation(): {
+  accept: (invitationId: string) => Promise<string>
+  accepting: boolean
+} {
+  const { sdk, resetAuthenticationToken } = useContext(LiteflowContext)
+  const [accepting, setAccepting] = useState(false)
+
+  const accept = useCallback(
+    async (invitationId: string) => {
+      try {
+        setAccepting(true)
+        // Ensure that there is no token in the context
+        resetAuthenticationToken()
+        const { acceptInvitation } = await sdk.AcceptInvitation({
+          id: invitationId,
+        })
```

Comment:
Can we just use plain graphql query?
Or maybe move this query directly to the starter kit?
There is no need to use sdk here.
Sdk is the issue here.
I've moved this query directly to the starter kit and used normal query and it worked without a problem.
As this is a no log in required process, should be better to have it on login page only?

Reply:
By moving it directly to the starter kit, you benefitted from the fact that the starter kit graphql client doesn't have an issue with the switch of account, while the initialization of the SDK seems to have a problem with it.
Resetting the token is not the solution here as this function should not control the authentication flow, but we need to ensure that when it's used, the actual SDK is reset correctly and not authentified